libraries/resource_systemd_mount.rb

```diff
@@ -0,0 +1,21 @@
+require_relative 'resource_systemd_unit'
+require_relative 'systemd_mount'
+
+class Chef::Resource
+  class SystemdMount < Chef::Resource::SystemdUnit
+    self.resource_name = :systemd_mount
+    provides :systemd_mount
+
+    def unit_type(_ = nil)
+      :mount
+    end
+
+    Systemd::Mount::OPTIONS.each do |option|
+      attribute option.underscore.to_sym, kind_of: String, default: nil
```

Comment:
totally. i figured this is a nice way to get bootstrapped really quickly, and then we can go through at our leisure and cherry pick important options to enhance for each resource type. i couldn't decide initially if i even wanted to enforce `String` types, but it was the easiest way to make sure the arguments supported `to_s` until we go do that extra enhancement.